scipy/linalg/tests/test_lapack.py

```diff
@@ -1645,3 +1645,50 @@ def test_getc2_gesc2():
         else:
             assert_array_almost_equal(desired_cplx.astype(dtype),
                                       x/scale, decimal=4)
+
+
+def test_gttrf_gttrs():
+    np.random.seed(42)
+    n = 10
+    for index, dtype in enumerate(DTYPES):
+        atol = 10**-(np.finfo(dtype).precision-1)
+        if index < 2:
+            du = np.random.rand(n-1)
+            d = np.random.rand(n)
+            dl = np.random.rand(n-1)
+        else:
+            du = np.random.rand(n-1) * 1j
+            d = np.random.rand(n) * 1j
+            dl = np.random.rand(n-1) * 1j
+
+        diag_cpy = np.array([dl.copy(), d.copy(), du.copy()])
+
+        A = np.diag(d) + np.diag(dl, -1) + np.diag(du, 1)
+        x = np.random.rand(n)
+        b = A @ x
+
+        dgttrf = get_lapack_funcs('gttrf', dtype=dtype)
+        dgttrs = get_lapack_funcs('gttrs', dtype=dtype)
+
+        _dl, _d, _du, du2, ipiv, info = dgttrf(dl, d, du)
+
+        np.testing.assert_allclose(dl, diag_cpy[0], atol=atol)
+        np.testing.assert_allclose(d, diag_cpy[1], atol=atol)
+        np.testing.assert_allclose(du, diag_cpy[2], atol=atol)
+        
+        b_cpy = b.copy()
+        x_dgttrs, info = dgttrs(_dl, _d, _du, du2, ipiv, b)
+        np.testing.assert_allclose(x, x_dgttrs, atol=atol)
+        np.testing.assert_allclose(b, b_cpy, atol=atol)
+        
+
+        dl_malformatted = np.append(dl, [1])
+        d_malformatted = np.append(d, [1])
+        du_malformatted = np.append(du, [1])
+        with assert_raises(ValueError):
+            dgttrf(dl_malformatted, d, du)
+        with assert_raises(ValueError):
+            dgttrf(dl, d_malformatted, du)
+        with assert_raises(ValueError):
+            dgttrf(dl, d, du_malformatted)
```

Comment:
Let's also see what happens when n = 2. @Kai-Striega has explicitly included a check in the wrapper:
`integer intent(hide),   depend(d), check(len(d) > 2) :: n = len(d)`
We should make sure it's working.

Reply:
It does does reject it and throws a nasty error that I can't seem to catch even with the most general: `assert_raises(Exception)`. <a href="https://ibb.co/NZ85GM1"><img src="https://i.ibb.co/Qr2Lx48/Screen-Shot-2019-12-15-at-3-33-49-PM.png" alt="Screen-Shot-2019-12-15-at-3-33-49-PM" border="0"></a>

Reply:
We cannot have n = 2 to ensure `du2`, an array of dimension ``n-2``, keeps a positive dimension.  You should still be able to catch the exception. Could you look at the test for `?tzrzf` - this tests a similar case for a different function, does that test pass for you? 

@mdhaber As an aside I do think we should expose this as some kind of more specialised Exception Class. From a user's perspective having to catch ``Exception``s is usually not the best way to go.

Reply:
You're right, it is catchable. Must have been a type with what I had before.